netket/hilbert/doubled_hilbert.py

```diff
@@ -0,0 +1,121 @@
+from .abstract_hilbert import AbstractHilbert
+from ..graph import DoubledGraph as _DoubledGraph
+
+import numpy as _np
+from netket import random as _random
+
+
+class PyDoubledHilbert(AbstractHilbert):
+    r"""Superoperatorial hilbert space for states living in the
+    tensorised state H\otimes H, encoded according to Choi's isomorphism."""
+
+    def __init__(self, hilb):
+        """Superoperatorial hilbert space for states living in the
+           tensorised state H\otimes H, encoded according to Choi's isomorphism.
+
+        Args:
+            hilb: the hilbrt space H.
+
+        Examples:
+            Simple superoperatorial hilbert space for few spins.
+
+           >>> from netket.graph import Hypercube
+           >>> from netket.hilbert import Spin, DoubledHilbert
+           >>> g = Hypercube(length=5,n_dim=2,pbc=True)
+           >>> hi = Spin(graph=g, s=0.5)
+           >>> hi2 = DoubledHilbert(hi)
+           >>> print(hi2.size)
+           50
+        """
+        doubled_graph = _DoubledGraph(hilb.graph)
```

Comment:
I think that this should happen only if the underlying hilbert has a graph, hilb.graph is not None. I.e. in the new abstract class for hilbert spaces we shouldn't assument that all hilbert spaces necessarily have a graph, I think